### DISCUSSION TITLE
Activity Pub, or Matrix protocol federation support for self-hosters?

### DISCUSSION THREAD

**Discussion by marek-lach** [stoatchat/discussions#125] [Feature Suggestions]
It'd fit perfectly with **Revolt** to implement the **[ActivityPub protocol](https://www.w3.org/TR/activitypub)**, or perhaps the [Matrix protocol](https://matrix.org/docs/spec), so that when people self-host it, they can _federate_ (i.e. see each-other) across the various servers, for example chatting in the form of _@username@server1.com_ writing to a _@username@server2.com_ .

That would be cool.

**Comment by oleggtro** [comment]
somewhat related to #103

**Comment by Zomatree** [comment]
We dont plan on adding federation, see here for the reasons why https://developers.revolt.chat/faq/federation

**Reply by oleggtro** [reply]
makes sense i guess

**Reply by da3dsoul** [reply]
The link is dead. Not sure if this is identical. https://developers.revolt.chat/faq.html

It looks like it has been updated following this discussion, as it might have alleviated some of the concerns

**Reply by DeclanChidlow** [reply]
https://developers.revolt.chat/faq.html#admonition-does-revolt-have-federation

**Comment by minkanjin** [comment]
I'm thinking actually replacing the Revolt protocols with Matrix protocols. That would fix the problem of not enough manpower. You would no longer have to maintain your own protocol, since most of the work would be done by the Matrix people (you just add the Revolt bits).

But I get why this is not an easy task and I'm not sure if Matrix solved the privacy problems.

**Reply by minkanjin** [reply]
Maybe there is another protocol that has most of the features that Revolt needs, e.g. ActivityPub or Mattermost. I guess some research would be needed.

**Reply by Mannshoch** [reply]
Ever considered to participate in the Matrix Protocol development. At least to check out the possibility? Or use the Matrix Protocoll but enhance it with own feature. 

I don't see the need that a Matrix Client need all possibilities and Feature Revolt has. I see no Problem if matrix user get a message like "this message could not be viewed with Matrix clients"

In other way It would be nice if the Revolt Client support Matrix server. And also here I see no Problem if features are grayed out with the Information. "Matrix server do not support this feature"

Even a basic support of matrix allows to see how the ecosystem move and what the next useful steps could be. Hey, there is even a possibility that even matrix user use Revolt with a reduced capability for Matrix because the app is still better than element. And you have a possibility to use one app for two ecosystems. I would consider it.

**Reply by DeclanChidlow** [reply]
@Mannshoch https://developers.revolt.chat/faq.html#admonition-does-revolt-have-federation

**Reply by Mannshoch** [reply]
Yes, I have read it before I wrote my text and I support the "MUST" section.

**Comment by erlend-sh** [comment]
There’s an emerging protocol that Revolt folks may wanna take a closer look at: [Polyproto](https://docs.polyphony.chat/Overviews/core/). It’s modeled after Discord, so it should be a lot easier to implement than the likes of Matrix or ActivityPub.

I’ve written about why I think federation is important in the context of chat applications here (in which ‘Matrix’ could easily be substituted for ‘Polyproto’):
- https://blog.commune.sh/beyond-discord/
- https://blog.commune.sh/federated-webrings/
- https://blog.erlend.sh/weird-netizens

**Comment by da3dsoul** [comment]
Why was this closed as resolved? Is this implemented? Is there more recent and relevant discussion somewhere else? I just got an email saying it's resolved with no context.

**Reply by DeclanChidlow** [reply]
Neither ActivityPub nor Matrix federation are planned.
